tls/s2n_client_hello.c

```diff
@@ -299,11 +321,38 @@ int s2n_process_client_hello(struct s2n_connection *conn)
     return S2N_SUCCESS;
 }
 
+static S2N_RESULT s2n_client_hello_process_cb_response(struct s2n_connection *conn, int rc)
+{
+    if (rc >= 0) {
+        if (conn->config->client_hello_cb_mode == S2N_CLIENT_HELLO_CB_BLOCKING) {
+            if (rc) {
+                conn->server_name_used = 1;
+            }
+            return S2N_RESULT_OK;
+        }
+        if (conn->config->client_hello_cb_mode == S2N_CLIENT_HELLO_CB_NONBLOCKING) {
+            if (conn->client_hello.callback_async_done) {
+                return S2N_RESULT_OK;
+            }
+            conn->client_hello.callback_async_blocked = 1;
+            RESULT_BAIL(S2N_ERR_ASYNC_BLOCKED);
+        }
+    }
+    /* unknown mode | rc < 0*/
```

Comment:
`switch` statement without a `default` will cause a compilation error whenever new mode is added without modification to the code here.

Reply:
good call, done